contracts/SnappAuction.sol

```diff
@@ -127,17 +118,13 @@ contract SnappAuction is SnappBase {
                     encodeOrder(accountId, buyTokens[i], sellTokens[i], buyAmounts[i], sellAmounts[i])
                 )
             );
-        }        
-        uint currentNonce = standingOrderPlacements[accountId].standingOrderNonce;
-        uint newNonce = currentNonce;
+        }
         
-        if (auctionIndex > standingOrderPlacements[accountId].reservedAccountOrders[currentNonce].validFromAuctionIndex) {
-            standingOrderPlacements[accountId].reservedAccountOrders[currentNonce].validToAuctionIndex = auctionIndex - 1;
-            newNonce += 1;
-            standingOrderPlacements[accountId].standingOrderNonce = newNonce;
-            standingOrderPlacements[accountId].reservedAccountOrders[newNonce].validFromAuctionIndex = auctionIndex;
+        if (auctionIndex > prevValidFromAuctionIndex) {
+            standingOrderPlacements[accountId][prevValidFromAuctionIndex].validToAuctionIndex = auctionIndex - 1;
+            standingOrderPlacements[accountId][auctionIndex].validToAuctionIndex = MAX_UINT;
         }
-        standingOrderPlacements[accountId].reservedAccountOrders[newNonce].orderHash = orderHash;
+        standingOrderPlacements[accountId][auctionIndex].orderHash = orderHash;
```

Comment:
This looks like a weird/inconsistent change that does not reflect identical logic to the previous code (which used an apparently different value).

Reply:
In my head this makes sense :) 
the mapping was changed to 
accoutID => validFromAuctionIndex => StandingOrderBatch

And the new validity will be set here to (validityFromAuctionIndex= auctionId)

This all goes a little bit further than felix suggestion, as it basically removes the nonce and stores the validFrom only indirectly.

Reply:
Cool, just making sure!